app/src/main/java/ch/epfl/sdp/drone3d/model/mission/Parallelogram.kt

```diff
@@ -51,16 +51,10 @@ data class Parallelogram(val origin: Point, val dir1Span: Vector, val dir2Span:
      * Returns the equivalent parallelogram with base point closest to [point]
      */
     fun getClosestEquivalentParallelogram(point: Point): Parallelogram {
+
         val vertices = getVertices()
-        var closestOriginIndex = 0
-        var smallestDistance =  distance(point,vertices[0])
-        for (i in 1 until vertices.size){
-            val distance = distance(point,vertices[i])
-            if(distance<smallestDistance){
-                smallestDistance = distance
-                closestOriginIndex = i
-            }
-        }
+        val closestOriginIndex = vertices.withIndex().minByOrNull{ (_, vertex) -> distance(point, vertex) }?.index ?: 0
```

Comment:
Great code refractoring ! I'm very impress by your skills

Reply:
Thank you, it all came from me